common/xfer/pipes.go

```diff
@@ -27,7 +27,16 @@ type pipe struct {
 	onClose         func()
 }
 
-// NewPipe makes a new... pipe.
+// NewPipeFromEnds makes a new pipe specifying its ends
+func NewPipeFromEnds(local io.ReadWriter, remote io.ReadWriter) Pipe {
```

Comment:
> needs to be broken down into multiple types for different uses

AFAICT pipes are always created and passed around as a `Pipe` interface so I am not sure what's the benefit of creating new types if they al implement `Pipe`.